Sensors crash when using `orjson` instead of `json`

## Checklist

- [x] I have included information about relevant versions
- [x] I have verified that the issue persists when using the `master` branch of Faust.

## Steps to reproduce

When using `orjson` instead of built-in `json`, the encoding behavior is different about dict keys.
I think the `orjson` backend is less tested because when using sensors, the call crashes:

config:
```python
app.web.blueprints.add("/stats/", "faust.web.apps.stats:blueprint")
```

When calling `/stats` with the browser:

```pytb
[2023-02-20 17:22:42,831] [64185] [ERROR] Error handling request 
Traceback (most recent call last):
  ...
  File "/home/elrik/Info/SecretSauce/cat-walk/venv/lib/python3.10/site-packages/faust/utils/json.py", line 184, in dumps
    return json_dumps(
TypeError: Dict key must be str
```

## Actual behavior

```python
>>> import json
>>> json.dumps({200: 1})  # Faust sensor http code counter
'{"200": 1}'
>>>  import orjson
>>> orjson.dumps({200: 1})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Dict key must be str
```

## Easy Fix:
If we want (almost) the same behavior between both, there is an easy fix with `OPT_NON_STR_KEYS`: https://github.com/ijl/orjson#opt_non_str_keys

```python
>>>  import orjson
>>> orjson.dumps({200: 1}, option=orjson.OPT_NON_STR_KEYS)
b'{"200":1}'
```


## Full traceback

```pytb
[2023-02-20 17:22:42,831] [64185] [ERROR] Error handling request 
Traceback (most recent call last):
  File "/home/elrik/Info/SecretSauce/cat-walk/venv/lib/python3.10/site-packages/aiohttp/web_protocol.py", line 433, in _handle_request
    resp = await request_handler(request)
  File "/home/elrik/Info/SecretSauce/cat-walk/venv/lib/python3.10/site-packages/aiohttp/web_app.py", line 504, in _handle
    resp = await handler(request)
  File "/home/elrik/Info/SecretSauce/cat-walk/venv/lib/python3.10/site-packages/aiohttp/web_middlewares.py", line 117, in impl
    return await handler(request)
  File "/home/elrik/Info/SecretSauce/cat-walk/venv/lib/python3.10/site-packages/aiohttp/web_middlewares.py", line 108, in impl
    return await handler(request)
  File "/home/elrik/Info/SecretSauce/cat-walk/venv/lib/python3.10/site-packages/faust/web/drivers/aiohttp.py", line 246, in _dispatch
    return await handler(request)
  File "/home/elrik/Info/SecretSauce/cat-walk/venv/lib/python3.10/site-packages/faust/web/views.py", line 102, in __call__
    return await self.dispatch(request)
  File "/home/elrik/Info/SecretSauce/cat-walk/venv/lib/python3.10/site-packages/faust/web/views.py", line 118, in dispatch
    response = await method(cast(Request, request), **kwargs)
  File "/home/elrik/Info/SecretSauce/cat-walk/venv/lib/python3.10/site-packages/faust/web/apps/stats.py", line 31, in get
    return self.json({f"Sensor{i}": s.asdict() for i, s in enumerate(self.app.sensors)})
  File "/home/elrik/Info/SecretSauce/cat-walk/venv/lib/python3.10/site-packages/faust/web/views.py", line 248, in json
    return self.web.json(
  File "/home/elrik/Info/SecretSauce/cat-walk/venv/lib/python3.10/site-packages/faust/web/drivers/aiohttp.py", line 181, in json
    payload: Any = _json.dumps(value)
  File "/home/elrik/Info/SecretSauce/cat-walk/venv/lib/python3.10/site-packages/faust/utils/json.py", line 184, in dumps
    return json_dumps(
TypeError: Dict key must be str
```

# Versions

* Python version: CPython 3.10.8 (Linux x86_64) + uvloop
* Faust version:  0.10.3

--- a/faust/utils/json.py
+++ b/faust/utils/json.py
@@ -175,6 +175,7 @@ if orjson is not None:  # pragma: no cover
         return json_dumps(
             obj,
             default=on_default,
+            option=orjson.OPT_NON_STR_KEYS,
         )
 
     def loads(s: str, json_loads: Callable = orjson.loads, **kwargs: Any) -> Any:

--- a/tests/unit/serializers/test_codecs.py
+++ b/tests/unit/serializers/test_codecs.py
@@ -32,9 +32,13 @@ def test_interface():
     assert s.__or__(1) is NotImplemented
 
 
-@pytest.mark.parametrize("codec", ["json", "pickle", "yaml"])
-def test_json_subset(codec: str) -> None:
-    assert loads(codec, dumps(codec, DATA)) == DATA
+@pytest.mark.parametrize("codec", ["pickle", "yaml"])
+def test_subset(codec: str) -> None:
+    if codec == "json":
+        # special exception for json since integers can be serialized
+        assert loads(codec, dumps(codec, DATA)) == {"a": 1, "b": "string", '1': 2}
+    else:
+        assert loads(codec, dumps(codec, DATA)) == DATA
 
 
 def test_missing_yaml_library() -> None:

--- a/tests/unit/serializers/test_codecs.py
+++ b/tests/unit/serializers/test_codecs.py
@@ -20,7 +20,7 @@ from faust.serializers.codecs import (
 )
 from faust.utils import json as _json
 
-DATA = {"a": 1, "b": "string"}
+DATA = {"a": 1, "b": "string", 1: 2}
 
 
 def test_interface():

--- a/tests/unit/serializers/test_codecs.py
+++ b/tests/unit/serializers/test_codecs.py
@@ -32,7 +32,7 @@ def test_interface():
     assert s.__or__(1) is NotImplemented
 
 
-@pytest.mark.parametrize("codec", ["pickle", "yaml"])
+@pytest.mark.parametrize("codec", ["json", "pickle", "yaml"])
 def test_subset(codec: str) -> None:
     if codec == "json":
         # special exception for json since integers can be serialized

--- a/tests/unit/serializers/test_codecs.py
+++ b/tests/unit/serializers/test_codecs.py
@@ -33,7 +33,7 @@ def test_interface():
 
 
 @pytest.mark.parametrize("codec", ["json", "pickle", "yaml"])
-def test_subset(codec: str) -> None:
+def test_json_subset(codec: str) -> None:
     if codec == "json":
         # special exception for json since integers can be serialized
         assert loads(codec, dumps(codec, DATA)) == {"a": 1, "b": "string", "1": 2}

--- a/tests/unit/serializers/test_codecs.py
+++ b/tests/unit/serializers/test_codecs.py
@@ -36,7 +36,7 @@ def test_interface():
 def test_subset(codec: str) -> None:
     if codec == "json":
         # special exception for json since integers can be serialized
-        assert loads(codec, dumps(codec, DATA)) == {"a": 1, "b": "string", '1': 2}
+        assert loads(codec, dumps(codec, DATA)) == {"a": 1, "b": "string", "1": 2}
     else:
         assert loads(codec, dumps(codec, DATA)) == DATA
 

